File: Project_Sourcecode_java/src/main/java/edu/iit/cs586/project/output/strategy/MakeDeposit/MakeDeposit2.java
package edu.iit.cs586.project.output.strategy.MakeDeposit;

import edu.iit.cs586.project.datastore.DS2;
import edu.iit.cs586.project.datastore.DataStore;

/**
 * @author CharmingLee <charminglee@jieztech.com>
 * @date 2022/5/2 23:41
 */
public class MakeDeposit2 implements MakeDeposit{
    @Override
    public void makeDeposit(DataStore ds) {
        float d = ((DS2)ds).temp_d;
        float b = ((DS2) ds).b;

        ((DS2) ds).b = d + b;
    }
}

File: Project_Sourcecode_go/output/strategy/makeDeposit/MakeDeposit2.go
package makeDeposit

import (
	"cs586_project/datastore"
	"fmt"
)

type MakeDeposit2 struct {
}

func (s *MakeDeposit2) MakeDeposit(ds interface{}) {
	ds2 := ds.(*datastore.DS2)
	d := ds2.Temp_d
	b := ds2.Balance
	ds2.Balance = d + b
	fmt.Printf("Make Deposit: %v\n", d)
}
